### PR TITLE
[WIP] Customisable @-menu

### LINKED ISSUE #51156: Customisable @-menu
# Is your feature request related to a problem? Please describe.
Currently there is no way to switch off or edit UI elements on the character screen like you can do with the sidebar. I think the option to customise @-menu would benefit players who play on smaller screens or don't like the default look of the menu.

# Describe the solution you'd like
Add an option to customise @-menu.

# Describe alternatives you've considered
None

### PR DESCRIPTION
**Pull request by Brambor**
#### Summary
Interface "Customizable @-menu"

#### Purpose of change

Fixes #51156, see it for more info and discussion.

#### Describe the solution

Changes:

- [x] SKILLS tab now has a scrollbar on the right & it's always visible.

Add the ability via a menu to customize @-menu in the following ways.

In this PR: 

- [ ] Number of columns.
- [ ] Width for each column.
- [ ] In which column to put individual windows & their order.
- [ ] Which windows not to display.
- [ ] Which horizontal / vertical dividers not to display.
- [ ] Maybe: Order of windows (in which order should be windows selected by pressing TAB).

In next PR:

- [ ] Which titles (EFFECTS, TRAITS, BIONICS, SKILLS...) not to display.
- [ ] Which individual lines in STATS not to display.

It should be similar to the sidebar customization but needs more to be truly useful.

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
@wapcaplet is unhardcoding sidebar widgets (https://github.com/CleverRaven/Cataclysm-DDA/pull/44683 and follow-ups https://github.com/CleverRaven/Cataclysm-DDA/pull/51182, https://github.com/CleverRaven/Cataclysm-DDA/pull/50945, https://github.com/CleverRaven/Cataclysm-DDA/pull/50910), I would guess you can utilize something from what was already done for sidebar for @-menu.

**Comment by Brambor** [comment]
> @wapcaplet is unhardcoding sidebar widgets (#44683 and follow-ups #51182, #50945, #50910), I would guess you can utilize something from what was already done for sidebar for @-menu.

I'm not sure but would like to hear his opinion.

I mostly wanted to select things to don't show & rearrange things, rather than make new functionality.

But then, I will have to store it somehow. Well, we will see.

**Comment by wapcaplet** [comment]
> I'm not sure but would like to hear his opinion.

Eventually I would like to "widget-ize" parts of the `@` screen for use in the sidebar, and/or use the new JSONizable widgets in more places throughout the UI, besides just the sidebar. For instance it would be great if we could display speed modiffiers or status effects from `@` in the main sidebar, and it would also be cool if we could allow players to mod their `@` screen layout at will, to do all the customizations you suggest and more (purely in JSON). But those things are still far ahead in my roadmap - I haven't got further than just thinking of the possibility, so it will probably be quite a while before it's ready for this part of the UI.

In other words, at this point, I don't think our work is likely to result in conflicts or duplication of effort. Your proposal looks good to me - having *any* customization to the `@` screen would be an improvement.

**Comment by Brambor** [comment]
@wapcaplet Thanks for your opinion I like to hear that. I don't have it as well thought out as you have. I don't want to use readable JSON but maybe it will come as side effect. If I get menu up and running, I would like to do everything in game, like options, clickity click, now it looks like shi... Anyway.

Having said that, I already see a reviewer having me reworking it. We will see. But I am at least a bit more hopeful now. The code base is quite huge and therefore intimidating.

**Comment by I-am-Erk** [comment]
I'd love to see this progress, but as it's been two months I am going to close this for now. Please reopen if you're around
